kitsune/sumo/views.py

```diff
@@ -156,3 +157,35 @@ def serve_cors(*args, **kwargs):
     from django.views.static import serve
 
     return serve(*args, **kwargs)
+
+
+@require_GET
+def translate_url(request):
+    """Find the correct translation URL for a given document.
+
+    Required parameters:
+    - current_slug
+    - current_locale
+    - target_locale
+    """
+    current_slug = request.GET.get("current_slug")
+    current_locale = request.GET.get("current_locale")
+    target_locale = request.GET.get("target_locale")
+
+    if not all([current_slug, current_locale, target_locale]):
+        return JsonResponse({"error": "Missing required parameters", "found": False}, status=400)
+
+    try:
+        current_doc = Document.objects.get(slug=current_slug, locale=current_locale)
+        parent_doc = current_doc.parent or current_doc
+        translation = Document.objects.filter(parent=parent_doc, locale=target_locale).first()
```

Comment:
Since the parent could be the "translation", I would change this to:
```python
translation = parent_doc if parent_doc.locale == target_locale else Document.objects.get(parent=parent_doc, locale=target_locale)
```
and then you can get rid of that `elif` clause below.